src/Microsoft.Extensions.CodeGenerators.Mvc/View/ViewGenerator.cs

```diff
@@ -62,77 +64,72 @@ public virtual IEnumerable<string> TemplateFolders
 
         public async Task GenerateCode([NotNull]ViewGeneratorModel viewGeneratorModel)
         {
-            ModelType model = ValidationUtil.ValidateType(viewGeneratorModel.ModelClass, "model", _modelTypesLocator);
-
             if (string.IsNullOrEmpty(viewGeneratorModel.ViewName))
             {
                 throw new ArgumentException("The ViewName cannot be empty");
             }
 
-            if (viewGeneratorModel.ViewName.EndsWith(Constants.ViewExtension, StringComparison.OrdinalIgnoreCase))
-            {
-                int viewNameLength = viewGeneratorModel.ViewName.Length - Constants.ViewExtension.Length;
-                viewGeneratorModel.ViewName = viewGeneratorModel.ViewName.Substring(0, viewNameLength);
-            }
-
             if (string.IsNullOrEmpty(viewGeneratorModel.TemplateName))
             {
                 throw new ArgumentException("The TemplateName cannot be empty");
             }
 
-            ModelType dataContext = ValidationUtil.ValidateType(viewGeneratorModel.DataContextClass, "dataContext", _modelTypesLocator, throwWhenNotFound: false);
-
-            // Validation successful
-            Contract.Assert(model != null, "Validation succeded but model type not set");
-
-            var appbasePath = _applicationEnvironment.ApplicationBasePath;
-            var outputFolder = String.IsNullOrEmpty(viewGeneratorModel.RelativeFolderPath)
-                ? Path.Combine(appbasePath)
-                : Path.Combine(appbasePath, viewGeneratorModel.RelativeFolderPath);
+            var outputPath = ValidateAndGetOutputPath(viewGeneratorModel, outputFileName: viewGeneratorModel.ViewName + Constants.ViewExtension);
```

Comment:
The behavior in VS is that view is created in the folder where the context menu is invoked, so I removed any assumptions about conventions in command line code and only using RelativePath if provided as the single options for modifying the output path. VS will pass the right value for this parameter.